oauth2/flowctx/encoding.go

```diff
@@ -0,0 +1,92 @@
+// Copyright © 2023 Ory Corp
+// SPDX-License-Identifier: Apache-2.0
+
+package flowctx
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/pkg/errors"
+
+	"github.com/ory/hydra/v2/jwk"
+)
+
+// Decode decodes the given string to a value.
+func Decode[T any](ctx context.Context, cipher *jwk.AEAD, encoded string) (*T, error) {
+	plaintext, err := cipher.Decrypt(ctx, encoded)
+	if err != nil {
+		return nil, err
+	}
+
+	rawBytes, err := gzip.NewReader(bytes.NewReader(plaintext))
+	if err != nil {
+		return nil, err
+	}
+	defer func() { _ = rawBytes.Close() }()
+
+	var val T
+	if err = json.NewDecoder(rawBytes).Decode(&val); err != nil {
+		return nil, err
+	}
+
+	return &val, nil
+}
+
+// Encode encodes the given value to a string.
+func Encode(ctx context.Context, cipher *jwk.AEAD, val any) (s string, err error) {
+	// Steps:
```

Comment:
Potentially: Consider using JWE / https://datatracker.ietf.org/doc/html/rfc7516

Reply:
I think fixing the encryption algorithm in code is best. Further, JWE does not support XChaChaPoly: https://www.iana.org/assignments/jose/jose.xhtml#web-signature-encryption-algorithms